hyperspy/tests/component/test_components.py

```diff
@@ -1,12 +1,20 @@
 import numpy as np
-
 from numpy.testing import assert_allclose
+import pytest
 
 import hyperspy.api as hs
 from hyperspy.models.model1d import Model1D
 from hyperspy.misc.test_utils import ignore_warning
 
 
+def _generate_parameters():
```

Comment:
To make the code more readable you could turn the ``_generate_parameters`` function into a global variable instead e.g:

```python
import itertools

TRUE_FALSE_2_TUPLE = [p for p in itertools.product((True, False), repeat=2)]
```